src/benchmark/benchmark_runner_main.cpp

```diff
@@ -0,0 +1,48 @@
+#include <cxxopts.hpp>
+
+#include "benchmark_runner.hpp"
+#include "benchmark_utils.hpp"
+#include "scheduler/current_scheduler.hpp"
+#include "scheduler/node_queue_scheduler.hpp"
+#include "scheduler/topology.hpp"
+#include "tpch/tpch_queries.hpp"
+#include "types.hpp"
+#include "utils/performance_warning.hpp"
+
+int main(int argc, char* argv[]) {
```

Comment:
Calling this without an argument segfaults for me. For some reason, I don't even have a stacktrace.

Reply:
that's really weird. i don't have this problem. i have one small fix regarding the filesystem include, where I messed up renaming. maybe this will fix it. otherwise, i have no idea tbh.

Reply:
tested on a second OS X machine. The help prints but then the program segfaults:

```
      --queries arg     Specify queries to run, either a single .sql file or
                        a directory with these files

Process 11599 stopped
* thread #1: tid = 0x20dc3, 0x00007fff97622c11 libc++abi.dylib`__dynamic_cast + 34, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=1, address=0x0)
    frame #0: 0x00007fff97622c11 libc++abi.dylib`__dynamic_cast + 34
libc++abi.dylib`__dynamic_cast:
->  0x7fff97622c11 <+34>: movq   (%rbx), %rax
    0x7fff97622c14 <+37>: movq   -0x10(%rax), %r13
    0x7fff97622c18 <+41>: addq   %rbx, %r13
    0x7fff97622c1b <+44>: movq   -0x8(%rax), %r14
(lldb) bt
* thread #1: tid = 0x20dc3, 0x00007fff97622c11 libc++abi.dylib`__dynamic_cast + 34, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=1, address=0x0)
  * frame #0: 0x00007fff97622c11 libc++abi.dylib`__dynamic_cast + 34
    frame #1: 0x0000000100006300 hyriseBenchmarkRunner`std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const& cxxopts::OptionValue::as<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >() const + 64
    frame #2: 0x0000000100003cea hyriseBenchmarkRunner`main + 5482
    frame #3: 0x00007fff9de1d5ad libdyld.dylib`start + 1
(lldb)
```

Reply:
ahh, it looks like i forgot to exit the program, when no queries and tables are specified. cxxopts just produces a segfault when i then access `result["queries"]` instead of a good error message.